Makefile

```diff
@@ -1864,6 +1864,9 @@ blob_file_garbage_test: $(OBJ_DIR)/db/blob/blob_file_garbage_test.o $(TEST_LIBRA
 blob_file_reader_test: $(OBJ_DIR)/db/blob/blob_file_reader_test.o $(TEST_LIBRARY) $(LIBRARY)
 	$(AM_LINK)
 
+blob_source_test: $(OBJ_DIR)/db/blob/blob_source_test.o $(TEST_LIBRARY) $(LIBRARY)
```

Comment:
Until our default build uses a shared library, additional test binaries for each "feature" leads to quadratic blow-up in build+link space and time (number of "features" times code size, which is proportional to features), because all of the RocksDB library is linked into each test binary. Only create new ones when there's a compelling reason.

Reply:
I guess we still want to keep this one since it's quite important to evaluate the blob cache. @ltamasi how do you think about this?

Reply:
I mean, we definitely want unit tests; I assume Peter's point was about creating a new unit test suite vs putting the new test cases in an existing one. FWIW, the Google C++ Style Guide seems to suggest (although from what I gather, doesn't mandate) using per-class test suites, see e.g. https://google.github.io/styleguide/cppguide.html#File_Names. That's also what I tend to do but I'm also fine with putting these tests in an existing suite.

Reply:
I keep the test cases in the new file since other test cases did the same way in the blob directory.